testing/integration/upgrade_test.go

```diff
@@ -273,7 +274,165 @@ func TestStandaloneUpgrade(t *testing.T) {
 	require.NoError(t, err, "error connecting client to agent")
 	defer c.Disconnect()
 
-	_, err = c.Upgrade(ctx, upgradeInputVersion.String(), "", false)
+	_, err = c.Upgrade(ctx, upgradeInputVersion.String(), "", false, false)
+	require.NoErrorf(t, err, "error triggering agent upgrade to version %q", upgradeInputVersion.String())
+
+	require.Eventuallyf(t, func() bool {
+		state, err := c.State(ctx)
+		if err != nil {
+			t.Logf("error getting the agent state: %v", err)
+			return false
+		}
+		t.Logf("current agent state: %+v", state)
+		return state.Info.Commit == expectedAgentHashAfterUpgrade && state.State == cproto.State_HEALTHY
+	}, 5*time.Minute, 1*time.Second, "agent never upgraded to expected version")
+
+	checkUpgradeWatcherRan(t, agentFixture)
+
+	version, err := c.Version(ctx)
+	require.NoError(t, err, "error checking version after upgrade")
+	require.Equal(t, expectedAgentHashAfterUpgrade, version.Commit, "agent commit hash changed after upgrade")
+}
+
+func TestStandaloneUpgrade_FailInitialCheck(t *testing.T) {
```

Comment:
I kinda wish that  this new test would be added after PR https://github.com/elastic/elastic-agent/pull/2955 because this will generate more conflicts :sweat: